src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -116,8 +127,30 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
                 // Write blob headers
                 await blobClient.SetHttpHeadersAsync(CreateHttpHeaders(artifactSettings), cancellationToken: token);
 
-                // Write blob metadata
-                await blobClient.SetMetadataAsync(artifactSettings.Metadata, cancellationToken: token);
+                Dictionary<string, string> mergedMetadata = new Dictionary<string, string>(artifactSettings.Metadata);
+
+                foreach (var metadataPair in artifactSettings.CustomMetadata)
+                {
+                    if (!mergedMetadata.ContainsKey(metadataPair.Key))
+                    {
+                        mergedMetadata[metadataPair.Key] = metadataPair.Value;
+                    }
+                    else
+                    {
+                        Logger.DuplicateKeyInMetadata(metadataPair.Key);
```

Comment:
Note for a future PR: planning to add in a prefix for our metadata (e.g. `DotnetMonitor_`), and then telling users to avoid setting environment variables to use that prefix. This will allow us to add more metadata without worrying about trampling existing user-provided metadata.